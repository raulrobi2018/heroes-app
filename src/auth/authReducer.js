import {types} from "../types/types";

export const authReducer = (state = {}, action) => {
    switch (action.type) {
        case types.login:
            return {
                // Retorna todo lo que venga en el payload
                ...action.payload,
                logged: true
            };
            break;

        case types.logout:
            return {
                logged: false
            };

        default:
            return state;
    }
};
