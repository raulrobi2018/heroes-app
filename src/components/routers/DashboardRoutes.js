import React from "react";
import {Redirect, Route, Switch} from "react-router-dom";
import {MarvelScreen} from "../marvel/MarvelScreen";
import {DcScreen} from "../dc/DcScreen";
import {Navbar} from "../ui/Navbar";
import {HeroScreen} from "../heroes/HeroScreen";

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />
            <div className="container mt-2">
                <Switch>
                    <Route exact path="/marvel" component={MarvelScreen} />
                    <Route exact path="/hero/:heroId" component={HeroScreen} />
                    <Route exact path="/dc" component={DcScreen} />

                    {/* Si no encuentra ning√∫n path, lo redirige a marvel */}
                    <Redirect to="/marvel" />
                </Switch>
            </div>
        </>
    );
};
