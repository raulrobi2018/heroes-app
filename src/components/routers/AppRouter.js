import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { LoginScreen } from "../login/LoginScreen";
import { DashboardRoutes } from "./DashboardRoutes";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        {/* A <Switch> looks through its children <Route>s and
              renders the first one that matches the current URL. */}
        <Switch>
          {/* exact para que machee la ruta exacta */}
          <Route exact path="/login" component={LoginScreen} />
          <Route path="/" component={DashboardRoutes} />
        </Switch>
      </div>
    </Router>
  );
};
