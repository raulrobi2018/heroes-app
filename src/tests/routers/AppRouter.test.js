import {mount, configure} from "enzyme";

import {AuthContext} from "../../auth/AuthContext";
import {AppRouter} from "../../components/routers/AppRouter";

//Esto es necesario para que no falle el mount en React 17
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
configure({adapter: new Adapter()});
// ------------------------------------------------------

describe("Testing AppRouter component", () => {
    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    };

    test("should display the LoginScreen if the user is not authenticated", () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test("should display the Marvel component if authenticated", () => {
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: "Jorge"
            }
        };
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect(wrapper.find(".navbar").exists()).toBe(true);
    });
});
