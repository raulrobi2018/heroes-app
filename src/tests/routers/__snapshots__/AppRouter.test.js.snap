// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AppRouter component should display the LoginScreen if the user is not authenticated 1`] = `
<div
  className="container"
>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
